### ISSUE TITLE
lastValidBlockHeight becomes undefined when serdes a legacy Transaction

### ISSUE THREAD

**Issue by charlesmaME** [solana-labs/solana-web3.js#3014]
## Overview

Our BE server fetches the `recentBlockhash` and `lastValidBlockHeight` via `connection.getLatestBlockhash` while creating a legacy Transaction. It then serializes the transaction and returns it to the FE on endpoint call. The FE then deserializes it, asks for user signature, and sends the transaction. The FE also calls `connection.confirmTransaction` with a `BlockhashWithExpiryBlockHeight` built using the `recentBlockhash` and `lastValidBlockHeight` from the deserialized Transaction, in order to save a `getLatestBlockhash` call.

We repeatedly ran into `TransactionExpiredBlockheightExceededError` when confirming the transaction though, immediately when attempting to `confirmTransaction`, because the `lastValidBlockHeight` of a legacy Transaction becomes `undefined` when you serialize, then deserialize it.

## Steps to reproduce

- Fetch latest blockhash
- Create legacy transaction using the latest blockhash data, being sure to include `lastValidBlockHeight`
- Console log the transaction's lastValidBlockHeight
- Serialize/deserialize the transaction
- Console log the deserialized transaction's lastValidBlockHeight

```
const blockhash = await CONN.getLatestBlockhashAndContext();
const tx = new anchor.web3.Transaction({
  feePayer: payer,
  blockhash: blockhash.value.blockhash,
  lastValidBlockHeight: blockhash.value.lastValidBlockHeight,
});
tx.add(
  ComputeBudgetProgram.setComputeUnitPrice({
    microLamports: priorityFeeMicroLamports,
  }),
  ...ixs,
);
console.log("pre-serdes", tx.lastValidBlockHeight);
const deserializedTx = anchor.web3.Transaction.from(tx.serialize({ requireAllSignatures: false }));
console.log("post-serdes", deserializedTx.lastValidBlockHeight);
```

## Description of bug

```
pre-serdes 259772183
post-serdes undefined
```

**Comment by steveluscher** [comment]
You're absolutely correct! Serializing a transaction is a lossy process. The serialized transaction contains the blockhash lifetime specifier, but no information about how long that blockhash will be valid.

Consider teaching your backend to send the transaction details to the frontend pre-serialization. This will be _much_ easier with the 2.0 line of `@solana/web3.js`, a Release Candidate version of which is coming out tomorrow.
